design-documents/service-isolation/detailed-architecture.md

```diff
@@ -0,0 +1,87 @@
+# Detailed architecture of multi-service Magento
+
+## Overview
+
+The goal of this document is to gather requirements, understand data flow and propose architecture for multi-service Magento application.
+
+Please make sure to get familiar with [high-level service isolation vision](../service-isolation.md) before proceeding.
+
+![Gateway](img/GraphQL%20BFF.png)
+
+The client (e.g. browser) is interacting with the system using GraphQL. The only component of the system exposed publicly is Storefront Gateway, which is responsible for routing, full response caching and perimeter-level authorization.
+
+Storefront Gateway routes incoming queries between PWA, GraphQL Engine and Storefront Authentication Gateway.
+
+PWA's main responsibility is to [render React application on the server-side](../frontend/server-side-rendering.md). Resulting HTML page with React application becomes cacheable on the server.
+
+GraphQL server parses GraphQL query, enriches the request (explained later) and proxies calls to the Storefront services. It also manages Customer context JWT. GraphQL schema is defined here and is extensible, query resolution is based on generic resolvers and configuration for field mapping to Storefront APIs.
```

Comment:
> calls to the Storefront services
Is it important to highlight that services are "Storefront"? The document is generic, and doesn't mention that it covers SF only scenarios.

Reply:
Initial intent was to create proposal for storefront BFF only. 
The document is not describing any admin scenarios, only storefront.